ibis/backends/dask/execution/temporal.py

```diff
@@ -0,0 +1,177 @@
+    ],
+    ops.IntervalSubtract: [
+        ((dd.Series, dd.Series), execute_timestamp_diff_sub_series_series)
+    ],
+    ops.IntervalMultiply: [
+        (
+            (timedelta_types, numeric_types + (dd.Series,)),
+            execute_interval_add_multiply_delta_series,
+        ),
+        (
+            (dd.Series, numeric_types + (dd.Series,)),
+            execute_interval_multiply_fdiv_series_numeric,
+        ),
+    ],
+    ops.IntervalFloorDivide: [
+        (
+            ((Timedelta, dd.Series), numeric_types + (dd.Series,),),
+            execute_interval_multiply_fdiv_series_numeric,
+        )
+    ],
+    ops.TimestampAdd: [
+        ((timestamp_types, dd.Series), execute_timestamp_add_datetime_series),
+        (
+            (dd.Series, timedelta_types),
+            execute_timestamp_interval_add_series_delta,
+        ),
+        ((dd.Series, dd.Series), execute_timestamp_interval_add_series_series),
+    ],
+    ops.TimestampSub: [
+        ((dd.Series, timedelta_types), execute_timestamp_sub_series_timedelta),
+        (
+            (timestamp_types, dd.Series),
+            execute_timestamp_diff_sub_datetime_series,
+        ),
+    ],
+    (ops.TimestampDiff, ops.TimestampSub): [
+        ((dd.Series, dd.Series), execute_timestamp_diff_sub_series_series)
+    ],
+    ops.TimestampDiff: [
+        ((dd.Series, timestamp_types), execute_timestamp_diff_series_datetime),
+        (
+            (timestamp_types, dd.Series),
+            execute_timestamp_diff_sub_datetime_series,
+        ),
+    ],
+    ops.DayOfWeekIndex: [((dd.Series,), execute_day_of_week_index_series)],
+    ops.DayOfWeekName: [((dd.Series,), execute_day_of_week_name_series)],
+    ops.Date: [((dd.Series,), execute_timestamp_date)],
+    ops.DateAdd: [
+        ((dd.Series, timedelta_types), execute_date_add),
+        ((timedelta_types, dd.Series), execute_date_add),
+        ((dd.Series, dd.Series), execute_date_add),
+        ((date_types, dd.Series), execute_date_add),
+        ((dd.Series, date_types), execute_date_add),
+    ],
+    ops.DateSub: [
+        ((date_types, dd.Series), execute_date_sub_diff),
+        ((dd.Series, dd.Series), execute_date_sub_diff),
+        ((dd.Series, timedelta_types), execute_date_sub_diff),
+    ],
+    ops.DateDiff: [
+        ((date_types, dd.Series), execute_date_sub_diff),
+        ((dd.Series, dd.Series), execute_date_sub_diff),
+        ((dd.Series, date_types), execute_date_sub_diff),
+    ],
+}
+register_types_to_dispatcher(execute_node, DASK_DISPATCH_TYPES)
+
+
+@execute_node.register(
+    ops.BetweenTime,
+    dd.Series,
+    (dd.Series, str, datetime.time),
+    (dd.Series, str, datetime.time),
+)
+def execute_between_time(op, data, lower, upper, **kwargs):
+    # TODO - Can this be done better?
+    indexer = (
+        (data.dt.time.astype(str) >= lower)
+        & (data.dt.time.astype(str) <= upper)
+    ).to_dask_array(True)
+
+    result = da.zeros(len(data), dtype=np.bool_)
```

Comment:
`len(data)` seems dangerous here since it will materialize data? Can we create a distributed array here by calling "apply" on the original dd.Series?

Reply:
Not sure. I'll add this to a "materialization to investigate" follow up issue (along with items you mentioned above).